dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,262 @@
+  parameters: z.object({
+    city: z.string().describe("The name of the city for which to retrieve the current time."),
+  }),
+  execute: ({city}) => {
+    return {status: 'success', report: \`The current time in \$\{ city \} is 10: 30 AM\`};
+  },
+});
+
+export const rootAgent = new LlmAgent({
+  name: 'hello_time_agent',
+  model: '${model}',
+  description: 'Tells the current time in a specified city.',
+  instruction: \`You are a helpful assistant that tells the current time in a city.
+                Use the 'getCurrentTime' tool for this purpose.\`,
+  tools: [getCurrentTime],
+});
+`.trim()
+
+interface AgentCreationOptions {
+  agentName: string;
+  model: string;
+  apiKey: string;
+  project: string;
+  region: string;
+  language: string;
+}
+
+async function getGcpProject(): Promise<string> {
+  if (process.env.GOOGLE_CLOUD_PROJECT) {
+    return process.env.GOOGLE_CLOUD_PROJECT;
+  }
+  try {
+    const stdout = execSync('gcloud config get-value project', { encoding: 'utf-8', stdio: 'pipe' });
+    return stdout.trim();
+  } catch (error) {
+    return '';
+  }
+}
+
+async function getGcpRegion(): Promise<string> {
+  if (process.env.GOOGLE_CLOUD_LOCATION) {
+    return process.env.GOOGLE_CLOUD_LOCATION;
+  }
+  try {
+    const stdout = execSync('gcloud config get-value compute/region', { encoding: 'utf-8', stdio: 'pipe' });
+    return stdout.trim();
+  } catch (error) {
+    return '';
+  }
+}
+
+async function generateAgentFolder(agentDir: string) {
+  if (!(await isFolderExists(agentDir))) {
+    return await createFolder(agentDir)
+  }
+  const overwriteFolderResponse: symbol | boolean = await select({
+      message: `Folder ${agentDir} already exists. Would you like to overwrite existing folder?`,
+      options: [
+        { label: 'Yes', value: true },
+        { label: 'No', value: false },
+      ],
+    },
+  );
+
+  if(isCancel(overwriteFolderResponse)) {
+    process.exit(0);
+  }
+
+  if (!overwriteFolderResponse) {
+    console.error(`Agent directory ${agentDir} already exists.`);
+    process.exit(0);
+  }
+  await removeFolder(agentDir);
+  await createFolder(agentDir);
+}
+
+async function generateFiles(options: AgentCreationOptions) {
+  const agentPath = path.join(dirname, options.agentName, `agent.${options.language}`);
+  const agentDir = path.dirname(agentPath);
+  const dotEnvPath = path.join(agentDir, '.env');
+
+  await saveToFile(agentPath, AGENT_TEMPLATE(options.model || 'gemini-2.5-flash'));
+
+  const lines = [];
+  if (options.apiKey) {
+    lines.push(`GOOGLE_API_KEY=${options.apiKey}`);
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=0`);
+  }
+  if (options.project) {
+    lines.push(`GOOGLE_CLOUD_PROJECT=${options.project}`);
+  }
+  if (options.region) {
+    lines.push(`GOOGLE_CLOUD_LOCATION=${options.region}`);
+  }
+  if (options.region && options.project) {
+    lines.push(`GOOGLE_GENAI_USE_VERTEXAI=1`);
+  }
+  await saveToFile(dotEnvPath, lines.join('\n'));
```

Comment:
can you please extract this to separate function (generateEnvFile) and use it here?